crates/bevy_window/src/window.rs

```diff
@@ -393,6 +407,26 @@ impl WindowResizeConstraints {
             );
             max_height = min_height;
         }
+
+        if (max_width.is_finite() && max_height.is_infinite())
+            || (max_width.is_infinite() && max_height.is_finite())
+        {
+            let (finite, finite_value, infinite) = if max_width.is_finite() {
+                let value = max_width;
+                max_width = f32::INFINITY;
+                ("width", value, "height")
+            } else {
+                let value = max_height;
+                max_height = f32::INFINITY;
+                ("height", value, "width")
+            };
+            warn!(
+                "The maximum {} is specified (value: {}) but the maximum {} is not (infinite). \
+                The {} constraint will be ignored, no maximum constraint will be set.",
+                finite, finite_value, infinite, finite
+            );
```

Comment:
Would setting these to the max value in this case work?

Reply:
I'm not sure what you mean, in the case of `max_width` being finite and `max_height` being `INFINITY`, set `max_height=max_width`, and vice-versa?

I'm not sure that's a good idea, there's no good solution in that case but I don't think when someone sets only one max constraint they want a square window. Also if we do that we need to change the order in this function so that a max constraint doesn't end up bellow a min constraint, and in that case we would have two warnings which would be kinda weird.

Btw I didn't test what happens when you set one max constraint to INFINITY in winit, I just did, it actually sets it to the min constraint :(

Reply:
I meant the max value that winit would accept I suppose.

Reply:
Ah. Let me check if I can just put `std::f32::MAX`.

Reply:
It doesn't work with `f32::MAX`, it doesn't work with `u32::MAX as f32`, I'm not really sure what is that maximum honestly.

Reply:
The value that seem to work is `((u32::MAX / 2) - 100) as f32` -_-'

Why this value? I'm not sure, I know the logical size is converted to physical size, and that physical size is a u32.
I guess at some point this u32 is converted to a i32, which makes it a negative value if it's above i32::MAX, and the max constraint becomes a negative one.
Why the `- 100`? If I don't put it it doesn't work, precision is lost because of `f32` but that shouldn't be a problem since it gives a lower value, but somehow it's "not enough", maybe because of window decorations? I have no idea I'm kinda lost.

I guess I'll try to code something to avoid this.

Reply:
Well I opened https://github.com/rust-windowing/winit/pull/2919 and I'll mark this PR blocked until I have an answer from Winit's team :x